### PR TITLE
Fix build error for mockito-android

### PR DESCRIPTION
**Pull request by tmurakami**
Fixed build error for `mockito-android` subproject.
Related to #922.

By this PR
- mockito-android will be unusable in Android 2.1 and below
- Android SDK is required for development

The workaround for #922 is effective only on Android 2.2 and later.
Because `AndroidJUnitRunner` can only be used with Android 2.2+.

According to [Android Dashboard](https://developer.android.com/about/dashboards/index.html), Android 2.1 and below seems to be ~1.0%.

Please note that it will not be possible to build `mockito-android` project on Travis CI because `travis.yml` has not been modified.
[Here](https://docs.travis-ci.com/user/languages/android/) is the guide to building an Android project in Travis CI.

### CODE REVIEW

**Comment by tmurakami** [comment]
I'm sorry. The license file was not included in archives.

File: gradle/mockito-core/javadoc.gradle [hunk 1]
```diff
@@ -66,8 +66,18 @@ javadoc {
...
         options.addStringOption('Xdoclint:none', '-quiet')
+
+        def minorVersion = System.getProperty("java.version") =~ /1.8.0_(.*)/
+
+        // Since JDK 8 version 121 scripts are not allowed in comments. However,
+        // earlier version throw an error if an unknown option is passed.
+        // Therefore only add this option if the JDK is actually correct.
+        if (minorVersion.size() > 0 && Integer.valueOf(minorVersion[0][1]) >= 121) {
+            options.addBooleanOption('-allow-script-in-comments', true)
+        }
```
Comment (TimvdLippe) [inline gradle/mockito-core/javadoc.gradle]: I think you need to rebase, since these changes are already merged.

File: gradle/pom-aar.gradle [hunk 2]
```diff
@@ -0,0 +1,45 @@
+publishing.publications.all {
+    //TODO github profile?
+    def devs = ['szczepiq:Szczepan Faber', 'bric3:Brice Dutheil', 'raphw:Rafael Winterhalter', 'TimvdLippe:Tim van der Lippe']
```
Comment (TimvdLippe) [inline gradle/pom-aar.gradle]: This is a lot of duplicate code. Can we reuse by first running the java-library pom.xml and then append the version of Mockito afterwards?

File: gradle/mockito-core/javadoc.gradle (hunk 1 above)
Comment (tmurakami) [inline gradle/mockito-core/javadoc.gradle]: Sorry. I had rebased.
Please point out if there are any problems.

**Comment by TimvdLippe** [comment]
I rebased `android-bundle` on `release/2.x`. Sadly you have to rebase again :sob: but it should be good to go now!

File: gradle/pom-aar.gradle (hunk 2 above)
Comment (tmurakami) [inline gradle/pom-aar.gradle]: Fixed pom-aar.gradle to reuse pom.gradle.

**Comment by tmurakami** [comment]
I have made a very ugly commit history.
I'm sorry...

**Comment by raphw** [comment]
Don*t worry, we squash it before we merge it. Its good to have steps during editing.

**Comment by bric3** [comment]
Still catching up...
@tmurakami what is wrong with this PR according to this https://github.com/mockito/mockito/issues/922#issuecomment-281696476 ?

**Comment by tmurakami** [comment]
@bric3 I'm sorry for the confusion.

This PR makes it impossible to execute the instrumentation test with `class` option as follows:
```
$ adb shell am instrument -w -e class foo.bar.BazTest ...
```

On Android Studio (and probably IntelliJ IDEA), the specified instrumentation test executes the above command, but an error occurs.

**Comment by bric3** [comment]
OK. Note I'm a total noob regarding android de tools. So I don't know what this change made, there's a new _aar_ artifact I see.

OK I missed this information `merged commit cc18d06 into mockito:android-bundle`

**Pull request merged**